definition_validator/validator.cc

```diff
@@ -530,9 +530,57 @@ void validateUnsatisfiedRequiredAncestors(core::Context ctx, const core::SymbolR
     }
 }
 
+void validateUnsatisfiableClassAncestors(core::Context ctx, const core::SymbolRef sym) {
+    auto data = sym.data(ctx);
+
+    vector<core::Symbol::RequiredAncestor> requiredClasses;
+    for (auto ancst : data->requiredAncestorsTransitive(ctx)) {
```

Comment:
This is somewhat expensive to compute, why not compute it once up front in `validateRequiredAncestors` and pass it through as a const reference?

Reply:
I would be happy to precompute it once the share it, but we also need it from `calls`. Any idea?

Reply:
Done in the new version at the end of the global pass 👍